samples/appendingEvents.go

```diff
@@ -0,0 +1,142 @@
+package samples
+
+import (
+	"context"
+	"log"
+
+	"github.com/EventStore/EventStore-Client-Go/client"
+	"github.com/EventStore/EventStore-Client-Go/messages"
+	"github.com/EventStore/EventStore-Client-Go/options"
+	"github.com/EventStore/EventStore-Client-Go/stream_position"
+	"github.com/EventStore/EventStore-Client-Go/streamrevision"
+)
+
+type TestEvent struct {
+	id            string
+	importantData string
+}
+
+func AppendToStream(db *client.Client) {
+	// region append-to-stream
+	data := TestEvent{
+		id:            "1",
+		importantData: "some value",
+	}
+
+	event, err := messages.NewJsonProposedEvent("some-event", data)
+
+	if err != nil {
+		panic(err)
+	}
+
+	options := options.AppendToStreamOptionsDefault()
+
+	result, err := db.AppendToStream(context.Background(), "some-stream", &options, event)
```

Comment:
Does `AppendToStreamOptions` need to be pointer in `AppendToStream` method? Cosmetics, but can make API nicer (and pointers are evil because people tend to forget to do nil checks :smiling_imp: )

Reply:
I really dislike pointers for that reason as well. However a few of the option parameter types used for adjacent methods - such as `PersistentAllSubscriptionOptions` - are a bit heavy and won't be deep copied when passed by value, so it makes sense to pass them by pointer. Because of that it's probably best to use pointers for all the similar `options` arguments for consistency.

Reply:
> are a bit heavy and won't be deep copied when passed by value, so it makes sense to pass them by pointer.

I'm not sure if it will be deep copied - Go is good at re-using values if they are the same in memory. I'm also sure that 
from performance point of view it is way faster to allocate this options thousand of times than doing a single network I/O call :wink: I think that better API should be a priority.

Reply:
I'm sure performance will be fine, in fact I initially also thought these should be passed by value. However I recalled using several [Go libraries](https://github.com/hashicorp/nomad/blob/main/api/api.go#L432) that [passed pointers](https://pkg.go.dev/github.com/jackc/pgx/v4/pgxpool#ParseConfig) for arguments like this despite the danger of nil references, and the motivation that occurs to me is that dealing with pass by values can be a bit misleading as some fields in the copied value will still point to the same data as their source value.